File: Array/3033.Modify_the_Matrix/C++/3033.Modify_the_Matrix.cpp
class Solution {
public:
    vector<vector<int>> modifiedMatrix(vector<vector<int>> & matrix) 
    {
        //==============================================//
        // Input type:                                  //
        //   - vector<vector<int>> ref. (matrix vector) //
        // Return type:                                 //
        //   - vector<vector<int>> (result vector)      //
        //==============================================//

        /*Initialize*/
        ///// Length of rows, columns /////
        int len_rows = matrix.size(), len_cols = (matrix[0]).size();

        /*Greedy-based loop traversal*/
        for (int cols_idx = 0; (cols_idx < len_cols); (cols_idx++)) //Columns
        {
            int record_max_val = (-2); //Record maximun value

            vector<int> record_vect; //Record vector

            for (int rows_idx = 0; (rows_idx < len_rows); (rows_idx++)) //Rows
            {
                ///// Check if the current indexed-value matched conditions or not /////
                if ((matrix[rows_idx])[cols_idx] == (-1))
                {
                    record_vect.emplace_back(rows_idx); continue;
                }

                record_max_val = max(record_max_val, (matrix[rows_idx])[cols_idx]); //Keep updating, overwriting

            } //Rows

            for (const int & record_val: record_vect) //Whole
            {
                (matrix[record_val])[cols_idx] = record_max_val; //Keep updating, overwriting

            } //Whole

        } //Columns

        return matrix;
    }
};

File: Array/3033.Modify_the_Matrix/Python/3033.Modify_the_Matrix.py
class Solution(object):
    def modifiedMatrix(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: List[List[int]]
        """
        #===============================#
        # Greedy-based traversal method #
        #===============================#

        ############
        #Initialize
        ##### Length of rows, columns #####
        len_rows, len_cols = len(matrix), len(matrix[0])

        #############################
        #Greedy-based loop traversal
        for cols_idx in range(len_cols):
            record_max_val = (-2) #Record maximun value

            record_arry = [] #Record array

            for rows_idx in range(len_rows):

                ##### Check if the current indexed-value matched conditions or not #####
                if ((matrix[rows_idx])[cols_idx] == (-1)):
                    record_arry.append(rows_idx) #Keep updating/recording

                    continue

                record_max_val = max(record_max_val, (matrix[rows_idx])[cols_idx]) #Keep updating/overwriting

            for record_val in record_arry:
                (matrix[record_val])[cols_idx] = record_max_val #Keep updating/overwriting

        return matrix
